repose-aggregator/components/filters/valkyrie-authorization/src/main/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilter.scala

```diff
@@ -409,12 +429,32 @@ class ValkyrieAuthorizationFilter @Inject()(configurationService: ConfigurationS
   }
 
   override def isInitialized: Boolean = initialized
+
+  trait ValkyrieResult
+
+  case class UserInfo(tenantId: String, contactId: String) extends ValkyrieResult
+
+  case class UserPermissions(roles: Vector[String], devices: Vector[DeviceToPermission]) extends ValkyrieResult
+
+  case class DeviceToPermission(device: Int, permission: String)
+
+  case class ResponseResult(statusCode: Int, message: String = "") extends ValkyrieResult
+
 }
 
 object ValkyrieAuthorizationFilter {
   val CACHE_PREFIX = "VALKYRIE-FILTER"
-}
 
-case class ResponseCullingException(message: String, throwable: Throwable) extends Exception(message, throwable) {
-  def this(message: String) = this(message, null)
+  class ResponseCullingException(message: String, throwable: Throwable = null) extends Exception(message, throwable)
+
+  class InvalidJsonPathException(message: String, throwable: Throwable = null) extends ResponseCullingException(message, throwable)
+
+  class InvalidJsonTypeException(message: String, throwable: Throwable = null) extends ResponseCullingException(message, throwable)
+
+  class InvalidRegexException(message: String, throwable: Throwable = null) extends ResponseCullingException(message, throwable)
+
+  class InvalidCaptureGroupException(message: String, throwable: Throwable = null) extends ResponseCullingException(message, throwable)
+
+  class NonMatchingRegexException(message: String, throwable: Throwable = null) extends ResponseCullingException(message, throwable)
```

Comment:
Having them be case classes might be better, although then the ResponseCullingException has to be a trait. Which means we would have a generic way to catch this type of exception, but not a generic way to throw this type of exception. Which basically just means that I need to go and replace where we've done that with specific exceptions.

I might do this to clean things up.